src/Catty/PlayerEngine/AudioEngine/AudioEngine.swift

```diff
@@ -141,6 +143,12 @@ import Foundation
         subtree.setInstrument(instrument)
     }
 
+    func setTempo(formula: Formula) {
```

Comment:
Please do not use a `Formula` in the context of the `AudioEngine`, pass a number instead. A `Formula` needs to be evaluated using the `FormulaInterpreter` (available in the Brick's instruction). Calling `formula.formulaTree.value` is only giving the proper value in case the Formula consists of one raw number only, which is rarely the case (e.g., because using functions, nested formulas, sensors, variables, etc.).

Reply:
Also consider adding a test for this method.